library/x509_crt.c

```diff
@@ -1435,32 +1571,179 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
 }
 #endif /* MBEDTLS_FS_IO */
 
+/*
+ * OtherName ::= SEQUENCE {
+ *      type-id    OBJECT IDENTIFIER,
+ *      value      [0] EXPLICIT ANY DEFINED BY type-id }
+ *
+ * HardwareModuleName ::= SEQUENCE {
+ *                           hwType OBJECT IDENTIFIER,
+ *                           hwSerialNum OCTET STRING }
+ *
+ * NOTE: we currently only parse and use otherName of type HwModuleName,
+ * as defined in RFC 4108.
+ */
+int mbedtls_x509_get_other_name( const mbedtls_x509_sequence *subject_alt_name,
+                                 mbedtls_x509_san_othername *other_name )
+{
+    int ret;
+    size_t len;
+    unsigned char *p = subject_alt_name->buf.p;
+    const unsigned char *end = p + subject_alt_name->buf.len;
+    mbedtls_x509_buf cur_oid = { 0, 0, NULL };
+
+    if( subject_alt_name->buf.tag != ( MBEDTLS_ASN1_CONTEXT_SPECIFIC | 0 ) )
```

Comment:
Please see my other comment. I would prefer to use macros instead of hardcoding 0 to mean "the value of otherName" as it makes the code easier to read

Reply:
ok